packages/backend/src/modules/tvl2/AmountService.ts

```diff
@@ -0,0 +1,211 @@
+import { assert, Logger } from '@l2beat/backend-tools'
+import {
+  assertUnreachable,
+  EscrowEntry,
+  TotalSupplyEntry,
+  UnixTime,
+} from '@l2beat/shared-pure'
+import { Configuration } from '@l2beat/uif'
+import { partition } from 'lodash'
+
+import {
+  ERC20MulticallCodec,
+  NativeAssetMulticallCodec,
+} from '../../peripherals/multicall/codecs'
+import { MulticallClient } from '../../peripherals/multicall/MulticallClient'
+import {
+  MulticallRequest,
+  MulticallResponse,
+} from '../../peripherals/multicall/types'
+import { RpcClient } from '../../peripherals/rpcclient/RpcClient'
+import { AmountRecord } from './repositories/AmountRepository'
+
+export type AmountConfiguration = EscrowEntry | TotalSupplyEntry
+
+export interface AmountServiceDependencies {
+  readonly rpcClient: RpcClient
+  readonly multicallClient: MulticallClient
+  readonly erc20Codec: ERC20MulticallCodec
+  readonly nativeAssetCodec?: NativeAssetMulticallCodec
+  logger: Logger
+}
+
+interface Amount {
```

Comment:
is this type needed?

Reply:
To define return types. Helps a lot when pinpointing bug in TS.